libcontainer/nsenter/nsenter_go122.go

```diff
@@ -0,0 +1,6 @@
+//go:build go1.22
+
+package nsenter
+
+//#error "runc does not currently work properly with Go >=1.22. See <https://github.com/opencontainers/runc/issues/4233>."
```

Comment:
This error should happen only when it is linked with a specific version of glibc

Reply:
Didn't Go merge the patch to always return an error?

Reply:
They did not merge it yet, and it's marked as WIP.

@AkihiroSuda it is kind of hard to say which glibc version won't work with which go version, so we're disabling this entirely for now, to avoid a bunch of issues being reported against 1.2.0-rc1. We plan to fix this somehow before the final 1.2.0 (see this PR description), so this is a temporary stop-gap measure.

Reply:
But at least can we still allow musl?

Reply:
We can, though I suspect in theory we are using undefined behaviour by running Go in the child of a `clone(2)` (see `signal-safety(7)` -- we aren't supposed to run any arbitrary code in a fork child). Maybe musl works but there's no guarantee it won't break in some strange case.